core/deployment/src/main/java/io/quarkus/deployment/steps/ConfigBuildSteps.java

```diff
@@ -62,6 +63,16 @@ void systemOnlySources(BuildProducer<StaticInitConfigBuilderBuildItem> staticIni
         runTimeConfigBuilder.produce(new RunTimeConfigBuilderBuildItem(SystemOnlySourcesConfigBuilder.class.getName()));
     }
 
+    /**
+     * Disable external configuration sources parsing for native executables since they are not supported see
+     * https://github.com/quarkusio/quarkus/issues/41994
+     */
+    @BuildStep(onlyIf = NativeOrNativeSourcesBuild.class)
+    void nativeNoSources(BuildProducer<StaticInitConfigBuilderBuildItem> staticInitConfigBuilder,
+            BuildProducer<RunTimeConfigBuilderBuildItem> runTimeConfigBuilder) {
+        runTimeConfigBuilder.produce(new RunTimeConfigBuilderBuildItem(SystemOnlySourcesConfigBuilder.class.getName()));
```

Comment:
I believe this is going to fail in the native build tests because this will disable the discovery of the external `config` folder:

https://github.com/smallrye/smallrye-config/blob/d61a0d7430adbd919121437f34b93b05d2a56f85/implementation/src/main/java/io/smallrye/config/SmallRyeConfigBuilder.java#L197-L220

Reply:
Yup, that's not gonna work. Closing this and I will prepare another PR registering the necessary files as resources.

Reply:
`io.smallrye.config.PropertiesConfigSourceLoader#inFileSystem` is public, so you can probably add a specific builder just for the native image with the `inFileSystem` registration.